TodoList/Src/TodoList.Api.Services/DependencyResolvers/DependencyResolver.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web.Http.Dependencies;
+using TodoList.Contracts.DI;
+
+namespace TodoList.Api.Services.DependencyResolvers
+{
+    public class DependencyResolver : IDependencyResolver
+    {
+        public readonly ITodoListProvider Container;
+
+        public static readonly List<string> KnowUnregisteredServiceNames = new List<string>(
```

Comment:
At most `internal` 😛

Reply:
You have a typo in the name - `know` is an imperative form while you likely want to use the pasive - `known`…

Reply:
Each `Type` instance in .NET is heavily cached to improve speed - see [the `Note` here](https://docs.microsoft.com/en-us/dotnet/api/system.type?view=netframework-4.7.2) - so it is kind of equal for both `Type`s and `string`s to be stored in an array. Storing types brings the advantage of access to more complex operation and shortens the code, so I would personally go for `Type`s here…

Reply:
Why are you making it a `List` when you plan no addition in it? 😉 It would be best to make it a generic enumerable with as syntactically simple construct as possible inside:
```suggestion
        internal static readonly IEnumerable<Type> KnownUnregisteredServiceNames = new []
```